### PR TITLE
4 main api for team

### PR DESCRIPTION
**Pull request by rayzhang-xingyunzh**

### CODE REVIEW

File: routes/teamAPI.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+//query parameters
+//	required:
+//		id:String (idea)
+router.get('/:id',teamController.getTeamById);
+
+//body parameters
+//	required:
+//	optional:
+//		pageNum:Number (default:1)
```
Comment (williamyu-xingyunzh) [inline routes/teamAPI.js:25]: why default value is 1?
Does "1" mean the first page? If yes, please make the first page start from 0, API is for programers, not for end user.

File: routes/teamAPI.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+//		pageSize:Number (default:10)
+//		keyword:String (for name)
+router.post('/list',teamController.list);
+
+
+//query parameters
+//	required:
+//		name:String 
+router.post('/check/name',teamController.checkNameExist);
```
Comment (williamyu-xingyunzh) [inline routes/teamAPI.js:34]: Is name here a query parameter in the path? Does it need to be :name 
have you tested this API?

File: models/team.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+		type:Schema.Types.ObjectId,
+		ref:'User'
+	}],
+
+	lead:{
+		type:Schema.Types.ObjectId,
+		ref:'User'
+	},
+
+	state:String
```
Comment (williamyu-xingyunzh) [inline models/team.js:41]: Suggest using Integer, as it's easier to sort inside DB. Up to you.

**Review by williamyu-xingyunzh — APPROVED** [review]
Please ensure you tested all the submitted code. High testing coverage is important to quality. quality also means time.

File: routes/teamAPI.js (hunk 2 above)
Comment (rayzhang-xingyunzh) [reply routes/teamAPI.js:34]: It was wrong comments.fixed.

File: routes/teamAPI.js (hunk 1 above)
Comment (rayzhang-xingyunzh) [reply routes/teamAPI.js:25]: fixed

**Pull request merged**